Src/DependencyCollector/Shared.Tests/Implementation/SqlClientDiagnosticSourceListenerTests.cs

```diff
@@ -0,0 +1,434 @@
+
+            var afterExecuteEventData = new
+            {
+                OperationId = operationId,
+                Operation = "ExecuteReader",
+                ConnectionId = sqlConnection.ClientConnectionId,
+                Command = sqlCommand,
+                Statistics = sqlConnection.RetrieveStatistics(),
+                Timestamp = 2000000L
+            };
+
+            var now = DateTimeOffset.UtcNow;
+
+            this.fakeSqlClientDiagnosticSource.Write(
+                SqlClientDiagnosticSourceListener.SqlAfterExecuteCommand,
+                afterExecuteEventData);
+
+            var dependencyTelemetry = (DependencyTelemetry)this.sendItems.Single();
+
+            Assert.AreEqual(afterExecuteEventData.OperationId.ToString("N"), dependencyTelemetry.Id);
+            Assert.AreEqual(sqlCommand.CommandText, dependencyTelemetry.Data);
+            Assert.AreEqual("(localdb)\\MSSQLLocalDB | master", dependencyTelemetry.Name);
+            Assert.AreEqual("(localdb)\\MSSQLLocalDB | master", dependencyTelemetry.Target);
+            Assert.AreEqual(RemoteDependencyConstants.SQL, dependencyTelemetry.Type);
+            Assert.IsTrue((bool)dependencyTelemetry.Success);
+            Assert.AreEqual(1000000L, dependencyTelemetry.Duration.Ticks);
+            Assert.IsTrue(dependencyTelemetry.Timestamp.Ticks < now.Ticks);
+        }
+        
+        [TestMethod]
+        public void TracksCommandError()
+        {
+            var operationId = Guid.NewGuid();
+            var sqlConnection = new SqlConnection(TestConnectionString);
+            var sqlCommand = sqlConnection.CreateCommand();
+            sqlCommand.CommandText = "select * from orders";
+
+            var beforeExecuteEventData = new
+            {
+                OperationId = operationId,
+                Timestamp = 1000000L
+            };
+
+            this.fakeSqlClientDiagnosticSource.Write(
+                SqlClientDiagnosticSourceListener.SqlBeforeExecuteCommand,
+                beforeExecuteEventData);
+
+            var commandErrorEventData = new
+            {
+                OperationId = operationId,
+                Operation = "ExecuteReader",
+                ConnectionId = sqlConnection.ClientConnectionId,
+                Command = sqlCommand,
+                Exception = new Exception("Boom!"),
+                Timestamp = 2000000L
+            };
+
+            var now = DateTimeOffset.UtcNow;
+
+            this.fakeSqlClientDiagnosticSource.Write(
+                SqlClientDiagnosticSourceListener.SqlErrorExecuteCommand,
+                commandErrorEventData);
+
+            var dependencyTelemetry = (DependencyTelemetry)this.sendItems.Single();
+
+            Assert.AreEqual(commandErrorEventData.Exception.ToInvariantString(), dependencyTelemetry.Properties["Exception"]);
+            Assert.IsFalse(dependencyTelemetry.Success.Value);
+        }
+        
+        [TestMethod]
+        public void TracksConnectionOpenedError()
```

Comment:
Since we are NOT supposed to track Successfull connections, add a test to validate we don't track sucessfull Open.

Reply:
Done.

Reply:
Done.